### PR TITLE
Data fixes

### PR DESCRIPTION
**Pull request by cmalinmayor**
@pattonw do you think that is the proper way to deal with mismatches between the labels dtype (short) and the prediction dtype (float)?

### CODE REVIEW

**Comment by pattonw** [comment]
Ahh, no I don't think so. There are cases you want the data types to be different (e.g. categorical cross entropy I'm pretty sure wants labels, not one-hot encoded floats), so I would consider directly casting the datatypes to be bad practice.

If possible I think the best way to handle this is in the `DataLoader`, doing the correct data transformations. But since this is a small controlled example where we do want the dtypes to be the same I don't think it causes too much harm to leave it as is.

**Comment by cmalinmayor** [comment]
Okay, thanks @pattonw! I'll merge and then if it causes issues in certain situations at least we will know where to look.
